modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/error/ConstructionError.scala

```diff
@@ -11,9 +11,11 @@ sealed trait ConstructionError extends NoStackTrace
 object ConstructionError {
   implicit class ConstructionErrorOps(ce: ConstructionError) {
     def toRosettaError: RosettaError = ce match {
-      case InvalidPublicKey => RosettaError.InvalidPublicKey
+      case InvalidPublicKey     => RosettaError.InvalidPublicKey
+      case MalformedTransaction => RosettaError.MalformedTransaction
     }
   }
 }
 
 case object InvalidPublicKey extends ConstructionError
+case object MalformedTransaction extends ConstructionError
```

Comment:
I am wondering if we need some context here. Like:

```scala
case class MalformedTransaction(hex: Hex) extends ConstructionError
```

up to you

Reply:
@cngo-github that is the case of metadata. if we would need any context, we can pass it to "details" error field which can be specific for a given request/response

Reply:
@marcinwadon long story short `metadata` is for `200` and `details` is for 500 but the purpose is more or less the same.

Reply:
I cannot think of a good, non-obvious context I can provide for this right now. I will leave it as is.

Reply:
👍